common/src/Model/TexCoordSystem.cpp

```diff
@@ -180,6 +180,13 @@ namespace TrenchBroom {
                 actualOffset[yIndex] = +offset.y();
             }
 
+            // Flip offset direction when texture scale is negative
+            if (std::signbit(attribs.scale().x()) == 1) {
```

Comment:
I'd prefer `attribs.scale().x() < 0.0f` for better readability, we don't need the genericness or nan-handling of std::signbit